django/utils/decorators.py

```diff
@@ -30,42 +32,56 @@ def _multi_decorate(decorators, method):
     a single decorator or an iterable of decorators.
     """
     if hasattr(decorators, "__iter__"):
-        # Apply a list/tuple of decorators if 'decorators' is one. Decorator
-        # functions are applied so that the call order is the same as the
-        # order in which they appear in the iterable.
+        # If 'decorators' is a list or tuple, reverse it to apply in correct order.
         decorators = decorators[::-1]
     else:
+        # If it's a single decorator, make it a list.
         decorators = [decorators]
 
-    def _wrapper(self, *args, **kwargs):
-        # bound_method has the signature that 'decorator' expects i.e. no
-        # 'self' argument, but it's a closure over self so it can call
-        # 'func'. Also, wrap method.__get__() in a function because new
-        # attributes can't be set on bound method objects, only on functions.
-        bound_method = wraps(method)(partial(method.__get__(self, type(self))))
+    if iscoroutinefunction(method):
+        # If the method is asynchronous, define an async wrapper.
+        async def _async_wrapper(self, *args, **kwargs):
```

Comment:
The variable can still be `_wrapper` in both branches I think.

Reply:
The variable can be wrapper but I think making it sync and async helps in readability but will change if my point does not make sense.

Reply:
Yes, please.